frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java

```diff
@@ -1239,4 +1267,131 @@ private void updateVulnerabilityVerification(ResourceRequest request, ResourceRe
         PrintWriter writer = response.getWriter();
         writer.write(responseData.toString());
     }
+
+    private void serveComponentList(ResourceRequest request, ResourceResponse response) throws IOException, PortletException {
+        Map<String, Integer> componentParameter = getServeListMapFromRequest(request);
+        List<Component> componentList = getFilteredComponentList(request);
+        JSONArray jsonComponents = getComponentData(componentList, componentParameter);
+
+        JSONObject jsonResult = createJSONObject();
+        jsonResult.put(DATATABLE_RECORDS_TOTAL, componentList.size());
+        jsonResult.put(DATATABLE_RECORDS_FILTERED, componentList.size());
+        jsonResult.put(DATATABLE_DISPLAY_DATA, jsonComponents);
+
+        try {
+            writeJSON(request, response, jsonResult);
+        } catch (IOException e) {
+            log.error("Problem rendering RequestStatus", e);
+        }
+    }
+
+    public JSONArray getComponentData(List<Component> componentList, Map<String, Integer> componentParameters) {
+        int startIdx = componentParameters.get(DATATABLE_DISPLAY_START);
+        int listDisplayAmount = componentParameters.get(DATATABLE_DISPLAY_LENGTH);
+        int sortColumnIdx = componentParameters.get(DATATABLE_SORT_COLUMN);
+        boolean isAscending = isSortOrderAscending(componentParameters);
+
+        List<Component> sortedComponents = sortComponentList(componentList, sortColumnIdx, isAscending);
+        int count = listDisplayAmount == -1 ? sortedComponents.size() : startIdx + listDisplayAmount;
```

Comment:
in the case of `listDisplayAmount != -1` that should be `min(startIdx + listDisplayAmount, <number-of-elements-left>)` ?

Reply:
changed to

`return min(componentParameters.getDisplayStart() + componentParameters.getDisplayLength(), maxSize);`